rspace/src/main/scala/coop/rchain/rspace/ReportingRspace.scala

```diff
@@ -65,7 +66,18 @@ object ReportingRspace {
       m: Match[F, P, A],
       scheduler: ExecutionContext
   ): F[ReportingRspace[F, C, P, A, K]] =
-    Sync[F].delay(new ReportingRspace[F, C, P, A, K](historyRepository, AtomicAny(store)))
+    for {
+      report     <- Ref.of(Seq.empty[Seq[ReportingEvent]])
+      softReport <- Ref.of(Seq.empty[ReportingEvent])
+      reportingRSpace <- Sync[F].delay(
+                          new ReportingRspace[F, C, P, A, K](
+                            historyRepository,
+                            storeAtom = AtomicAny(store),
+                            report,
+                            softReport
+                          )
+                        )
+    } yield (reportingRSpace)
```

Comment:
No need for 3rd Sync[F].delay, you can go with
```scala
for {
      report     <- Ref.of(Seq.empty[Seq[ReportingEvent]])
      softReport <- Ref.of(Seq.empty[ReportingEvent])
    } yield new ReportingRspace[F, C, P, A, K](
      historyRepository,
      storeAtom = AtomicAny(store),
      report,
      softReport
    )
```

Reply:
Thank you, @nzpr  for your comment. I will take this into account in the future.